### PR TITLE
Changing iradon filters to match those of MATLAB and Octave

### PR DESCRIPTION
**Pull request by ASchuberth**
Original issue: https://github.com/scikit-image/scikit-image/issues/2853

## Description
The filters currently being used do not match the MATLAB or Octave filters.  Output of the current program gives:

oldfilters

------

The proposed changes give an output of:

newfilters

which matches MATLAB and Octave.  The source code of the rho_filter function in Octave was used to generate them.

------ 

These changes pass all the tests.  However, as I was researching these filters, I noticed that they are shown in this format:
paperfilters

For example, [rice.edu notes](https://www.clear.rice.edu/elec431/projects96/DSP/filters.html) and in this [paper](https://www.researchgate.net/publication/251915833_A_prototype_system_for_infrared_computed_tomography_for_image_reconstruction).  These are the same filters, but with np.fft.fftshift performed on them.  I feel like these are what we should use, but they fail the test_radon_transform tests.  Does anyone know why we use the other shifted filters instead of these?  Octave appears to do the same thing.  I haven't tested these filters with test phantoms yet to check the difference in output.

------- 
The current filters in this branch all in the same graph for reference:
allfilters
 

## Checklist
[It's fine to submit PRs which are a work in progress! But before they are merged, all PRs should provide:]
- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- [x] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [x] Gallery example in `./doc/examples` (new features only)
- [ ] Unit tests

[For detailed information on these and other aspects see [scikit-image contribution guidelines](http://scikit-image.org/docs/dev/contribute.html)]

## References
This will close issue #2853.
[Octave rho_filter Source Code](https://sourceforge.net/p/octave/image/ci/default/tree/inst/rho_filter.m#l130)
[Rice.edu Notes](https://www.clear.rice.edu/elec431/projects96/DSP/filters.html)

## For reviewers

(Don't remove the checklist below.)

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by matthew-brett** [comment]
Hi - just checking - "The source code of the rho_filter function in Octave was used to generate them." - you mean generate the figures and not the code right?  Otherwise we'd have a license problem.

**Comment by oachkatzlschwoaf123** [comment]
I referenced the rho_filter source code and edited the iradon function code in radon_transform.py to match the rho_filter code output.  Is that okay? Is there a reference in the code needed?  All of the figures here were generated from the original iradon code, the code from the OP of  issue #2853 , and edits that I made.

Thanks for catching that. I should have phrased that differently.

**Comment by oachkatzlschwoaf123** [comment]
Disregard what I asked earlier about the filters.  The way I have it coded is the way it needs to be.

File: skimage/.cache/v/cache/lastfailed [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (soupault) [inline skimage/.cache/v/cache/lastfailed]: The cache files must be removed.

File: skimage/transform/.cache/v/cache/lastfailed [hunk 2]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (soupault) [inline skimage/transform/.cache/v/cache/lastfailed]: This one also :)

**Comment by stefanv** [comment]
Can we add a test to ensure that this doesn't break again?  E.g., generate an impulse response of the filter, and verify that Shepp-Logan has two peaks, or peaks at the expected locations, etc.

**Comment by ASchuberth** [comment]
Yeah, I'll try to think of something.  Sorry for the late response.  I have been pretty busy, but I should have time next week to work on this.

**Comment by kczimm** [comment]
Any update on this @ASchuberth? I've avoided using scikit-image due to the image bias incurred from the wrong recon kernels. I would love to start using it once the kernel bug is corrected. Thanks in advance for this effort.

**Comment by stefanv** [comment]
@kczimm This PR is almost there, it just needs a test; is this something you can help with?

**Comment by kczimm** [comment]
@stefanv I can try to help. I've never contributed to this project. At the risk of exposing my naïveté, how do I execute the tests in the test_radon_transform.py file?

**Comment by soupault** [comment]
@kczimm see http://scikit-image.org/docs/dev/contribute.html#testing

**Comment by kczimm** [comment]
After some inspection, the kernels still have a bias problem. But this is no surprise to me because they were modeled from Octave and that implementation has the same bias problem but MATLAB does not. The following script reconstructs a uniform circular phantom and plots the line profile through the center.

```
from __future__ import print_function, division

import numpy as np
import matplotlib.pyplot as plt

from skimage.transform import iradon, radon

pixels = 512
xy = np.arange(-pixels/2, pixels/2) + 0.5
x, y = np.meshgrid(xy, xy)
image = x**2 + y**2 <= (pixels/4)**2

theta = np.linspace(0., 180., max(image.shape), endpoint=False)
sinogram = radon(image, theta=theta, circle=True)

reconstruction_fbp = iradon(sinogram, theta=theta, circle=True)
error = reconstruction_fbp - image

plt.plot(reconstruction_fbp[int(pixels/2), :])
plt.plot(image[int(pixels/2), :])
plt.show()
```

If you zoom in on the uniform region of the phantom or the background, there is an approximate ~0.5% bias. While the reconstruction looks qualitatively correct, it is biased to a non-trivial degree. The zero frequency of the kernels needs to be adjusted to preserve the DC component of the image.

bias

The ramp filter should be implemented in the spatial domain first as in Kak and Slaney p.72 (https://engineering.purdue.edu/~malcolm/pct/CTI_Ch03.pdf)

**Pull request closed without merging**